### PR TITLE
chore(crafter): Get #smithed.crafter:all from misode/mcmeta

### PR DESCRIPTION
**Pull request by edayot**
Get the item list from "[https://raw.githubusercontent.com/misode/mcmeta/{version}-registries/item/data.json](https://raw.githubusercontent.com/misode/mcmeta/%7Bversion%7D-registries/item/data.json)" (where version is the ctx.minecraft_version).

### CODE REVIEW

**Comment by RitikShah** [comment]
This is a clean idea, thanks for writing this up. I'll want to use the `Vanilla` class from `beet.contrib.vanilla` for this but the idea is good 👍🏿

**Comment by edayot** [comment]
Because the information is in the registries and not in the `data` folder, I haven't found a way with beet to extract the list.

**Comment by RitikShah** [comment]
Ah, this makes sense then. I think this will be a good solution.

File: smithed_libraries/plugins/crafter.py
Comment (vberlier) [inline smithed_libraries/plugins/crafter.py:1]: I'd suggest using beet's `cache.download()` method
```py
def beet_default(ctx: Context):
    cache = ctx.cache["smithed_crafter"]

    all_items_url = f"https://raw.githubusercontent.com/misode/mcmeta/{ctx.minecraft_version}-registries/item/data.json"
    all_items = ItemTag(source_path=cache.download(all_items_url))
    all_items.data = {"values": all_items.data}

    ctx.generate("smithed.crafter:all", all_items)
```

Comment (RitikShah) [reply smithed_libraries/plugins/crafter.py:1]: Very clean!

**Pull request merged**